gmso/formats/lammpsdata.py

```diff
@@ -960,10 +1229,10 @@ def _write_site_data(out_file, top, atom_style, base_unyts, cfactorsDict):
         out_file.write(
             atom_line.format(
                 index=i + 1,
-                moleculeid=site.molecule.number,
+                moleculeid=site.molecule.number + 1,  # index is 0-based in GMSO
                 type_index=unique_sorted_typesList.index(site.atom_type) + 1,
                 charge=base_unyts.convert_parameter(
-                    site.charge,
+                    site.atom_type.charge if site.atom_type else site.charge,
```

Comment:
is there a reason we revert the priority for charge here? Current the priority for site's charge is 
> manual site's charge > site's atom_type's charge
when `site.charge` is called, it will check if there is a manual value set for `site.charge`, if not it will go for `site.atom_type.charge` (if `site.atom_type` exist.

Reply:
we may want to discuss this in as a group also (priority of certain attribute)

Reply:
Agreed on the charge. I think I just changed this on a whim because something was causing tests to fail when site.charge was not set. Maybe there's a bug in checking in site.atom_type.charge. I think the current behavior should be the standard, so I'll figure out the cause of that bug and fix it and then revert this change. We can discuss more next Monday as well.